src/test/test_mempool.cc

```diff
@@ -120,9 +121,11 @@ TEST(mempool, vector_context)
 
 TEST(mempool, list_context)
 {
+  EXPECT_EQ(mempool::osd::containers(), 0u);
```

Comment:
Huh, nasty failures after trivial rebase. For instance:
```
/home/perf/src/ceph-rzarzynski/src/test/test_mempool.cc:124: Failure
      Expected: mempool::osd::containers()
      Which is: 1
To be equal to: 0u
      Which is: 0
```

Let's see if somebody really constructs a container in the `mempool::osd` space before the failure:
```
(gdb) break mempool::osd::pool_allocator::ctor
Breakpoint 4 at 0x7fffeef9730a: file /home/perf/src/ceph-rzarzynski/src/include/mempool.h, line 402.
(gdb) run
Starting program: /home/perf/src/ceph-rzarzynski/build/bin/unittest_mempool --gtest_filter=mempool.list_context
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
warning: File "/usr/local/lib64/libstdc++.so.6.0.24-gdb.py" auto-loading has been declined by your `auto-load safe-path' set to "$debugdir:$datadir/auto-load:/usr/share/gdb/auto-load".

Breakpoint 4, mempool::pool_allocator<(mempool::pool_index_t)12, PGPeeringEvent>::pool_allocator (force_register=true, this=0x7ffff79b9440 <mempool::osd::alloc_pg_peering_evt>)
    at /home/perf/src/ceph-rzarzynski/src/include/mempool.h:465
465         this->ctor(force_register,sizeof(T));
(gdb) bt
#0  mempool::pool_allocator<(mempool::pool_index_t)12, PGPeeringEvent>::pool_allocator (force_register=true, this=0x7ffff79b9440 <mempool::osd::alloc_pg_peering_evt>)
    at /home/perf/src/ceph-rzarzynski/src/include/mempool.h:465
#1  __static_initialization_and_destruction_0 (__initialize_p=1, __priority=65535) at /home/perf/src/ceph-rzarzynski/src/osd/PGPeeringEvent.cc:8
#2  _GLOBAL__sub_I_PGPeeringEvent.cc(void) () at /home/perf/src/ceph-rzarzynski/src/osd/PGPeeringEvent.cc:8
#3  0x00007ffff7dea4c3 in call_init (env=0x7fffffffda60, argv=0x7fffffffda48, argc=2, l=<optimized out>) at dl-init.c:82
#4  _dl_init (main_map=0x7ffff7ffe150, argc=2, argv=0x7fffffffda48, env=0x7fffffffda60) at dl-init.c:131
#5  0x00007ffff7ddc1aa in _dl_start_user () from /lib64/ld-linux-x86-64.so.2
#6  0x0000000000000002 in ?? ()
#7  0x00007fffffffdd7c in ?? ()
#8  0x00007fffffffddb6 in ?? ()
#9  0x0000000000000000 in ?? ()
(gdb) cont
Continuing.
[New Thread 0x7fffe90a5700 (LWP 5441)]
[New Thread 0x7fffe812e700 (LWP 5442)]
Note: Google Test filter = mempool.list_context
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from mempool
[ RUN      ] mempool.list_context
/home/perf/src/ceph-rzarzynski/src/test/test_mempool.cc:124: Failure
      Expected: mempool::osd::containers()
      Which is: 1
To be equal to: 0u
      Which is: 0
```

In `src/osd/PGPeeringEvent.cc`:
```cpp
MEMPOOL_DEFINE_OBJECT_FACTORY(PGPeeringEvent, pg_peering_evt, osd);
```

which expands to something like:
```cpp
namespace mempool { 
  namespace osd {
    pool_allocator<PGPeeringEvent> alloc_pg_peering_evt = {true};
  }
}
```

`unittest_mempool` links with `global` which in turn pulls `ceph-common`. Should  `PGPeeringEvent.cc` belong to `libcommon_files`?

Reply:
Moved on with a workaround (relative values). Still, I'm concerned whether `PGPeeringEvent.cc` should be really a part of the `libcommon_files`. @liewegas?